source/egg/math/Quat.cc

```diff
@@ -47,16 +47,23 @@ Vector3f Quatf::rotateVector(const Vector3f &vec) const {
     return res.v;
 }
 
+Vector3f Quatf::rotateVectorInv(const Vector3f &vec) const {
+    Quatf conj = conjugate();
+    Quatf res = conj * vec;
+    res *= *this;
+    return res.v;
+}
+
```

Comment:
I believe that this is not correct, given that associativity is not guaranteed for floats. The last component of the vector cross product (generated in `Quatf::operator*(Quatf)`) is not subtracted until after all other terms are added. This does seem to affect the final result, and actually desyncs frame 4 for me.

I believe this also applies to `rotateVector()`

Reply:
This is addressed in stage 1 branch